src/realm/object-store/audit.mm

```diff
@@ -1081,45 +1089,82 @@ void validate_metadata(std::vector<std::pair<std::string, std::string>>& metadat
 
 {
     util::CheckedLockGuard lock(m_mutex);
-    auto scope = std::make_shared<Scope>(Scope{m_metadata, std::string(activity)});
+    auto scope = std::make_shared<Scope>(Scope{0, m_metadata, std::string(activity)});
```

Comment:
shouldn't this be using `++m_scope_counter`? Otherwise, `0` could use an explanatory comment.

Reply:
The id is only used for looking things up in m_current_scopes (which is a vector and not a key-lookup data structure because it's nearly always going to have zero, one, or very occasionally two entries), so this scope's id is not used. I'll add a comment.